UnitTests/dnYara.UnitTests/ScanTests.cs

```diff
@@ -124,5 +125,47 @@ public void CheckMemoryNotMatchTest()
                 }
             }
         }
+
+        [Fact]
+        public void CheckIterateRulesTest()
+        {
+            string ruleText = @"rule foo: bar {
+                meta:
+                    bool_meta = true
+                    int_meta = 10
+                    string_meta = ""what a long, drawn-out thing this is!""
+                strings:
+                    $a = ""nml""
```

Comment:
I didn't include extraction of the `strings` map of the rule yet, though now that the pattern for iteration is in place it would be easy.